src/Microsoft.DotNet.Build.Tasks.Feed/CopyBlobDirectory.cs

```diff
@@ -74,55 +82,66 @@ public async Task<bool> ExecuteAsync()
-                            token = segment.ContinuationToken;
-                            sourceBlobs.AddRange(segment.Results);
+                            Log.LogError($"Target blob {targetBlob.Uri.ToString()} already exists.");
                         }
-                        while (token != null);
 
-                        // Ensure the source exists
-                        if (!SkipIfMissing && sourceBlobs.Count == 0)
+                        BlobUrlInfo specificSourceBlobUrlInfo = new BlobUrlInfo(blob.Uri);
+                        CloudBlob sourceBlob = sourceContainer.GetBlobReference(specificSourceBlobUrlInfo.BlobPath);
+                        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy
                         {
-                            Log.LogError($"No blobs found in {sourceUri}");
-                        }
-
-                        await Task.WhenAll(sourceBlobs.Select(async blob =>
+                            Permissions = SharedAccessBlobPermissions.Read,
+                            SharedAccessStartTime = null,
+                            SharedAccessExpiryTime = DateTimeOffset.Now.AddMinutes(30)
+                        };
+                        string sas = sourceBlob.GetSharedAccessSignature(policy);
+                        await targetBlob.StartCopyAsync(new Uri(blob.Uri + sas));
+
+                        var copyInProgessBlob = await targetContainer.GetBlobReferenceFromServerAsync(specificTargetBlobUrlInfo.BlobPath);
+                        while (copyInProgessBlob.CopyState.Status == CopyStatus.Pending)
                         {
-                            // Determine the relative URI for the target.  This works properly when the
-                            // trailing slash is left off of the source and target URIs.
-                            string relativeBlobPath = blob.Uri.ToString().Substring(sourceUri.Length);
-                            string specificTargetUri = GetCanonicalStorageUri(targetUri, relativeBlobPath);
-                            BlobUrlInfo specificTargetBlobUrlInfo = new BlobUrlInfo(specificTargetUri);
-                            CloudBlob targetBlob = targetContainer.GetBlobReference(specificTargetBlobUrlInfo.BlobPath);
-
-                            Log.LogMessage($"Merging {blob.Uri.ToString()} into {targetBlob.Uri.ToString()}");
-
-                            if (!Overwrite && await targetBlob.ExistsAsync())
-                            {
-                                Log.LogError($"Target blob {targetBlob.Uri.ToString()} already exists.");
-                            }
-                            
-                            await targetBlob.StartCopyAsync(blob.Uri);
-                        }));
-                    }
+                            await Task.Delay(TimeSpan.FromSeconds(30));
+                            copyInProgessBlob = await targetContainer.GetBlobReferenceFromServerAsync(specificTargetBlobUrlInfo.BlobPath);
```

Comment:
Consider some other loop construct to remove the duplicated call.

(I was also going to ask if this will ever time out, but I guess the SAS token will implicitly stop it after 30 mins.)

Reply:
I'm not positive it would.  I'll add a wait timeout and some additional error handling after that